pkg/util/push/otlp.go

```diff
@@ -87,6 +90,34 @@ func OTLPHandler(sourceIPs *middleware.SourceIPExtractor, push Func) http.Handle
 	})
 }
 
+func convertToPromTS(ctx context.Context, pmetrics pmetric.Metrics, cfg distributor.OTLPConfig, promoteResourceAttributes []string, logger log.Logger) ([]prompb.TimeSeries, error) {
+	promConverter := prometheusremotewrite.NewPrometheusConverter()
+	settings := prometheusremotewrite.Settings{
+		AddMetricSuffixes:         true,
+		PromoteResourceAttributes: promoteResourceAttributes,
+		DisableTargetInfo:         cfg.DisableTargetInfo,
+	}
+
+	var annots annotations.Annotations
+	var err error
+	if cfg.ConvertAllAttributes {
+		annots, err = promConverter.FromMetrics(ctx, convertToMetricsAttributes(pmetrics), settings)
+	} else {
+		annots, err = promConverter.FromMetrics(ctx, pmetrics, settings)
```

Comment:
We can call `overrides.PromoteResourceAttributes(userID)` here so we don't have to eagerly get the lists even if we don't convert all attributes.